File: api/src/main/java/com/panucci/mlp/dto/StartTrainingPayload.java
package com.panucci.mlp.dto;

public record StartTrainingPayload(
    String sessionId,
    String databaseName,
    int hiddenLayersNumber,
    String activationFunctionName,
    double learningRate,
    double stopError,
    int maxEpochs,
    TrainingEventOptions eventOptions
) {
    public StartTrainingPayload withSessionId(String sessionId) {
        return new StartTrainingPayload(
            sessionId,
            this.databaseName,
            this.hiddenLayersNumber,
            this.activationFunctionName,
            this.learningRate,
            this.stopError,
            this.maxEpochs,
            this.eventOptions
        );
    }
}

File: frontend/src/types/StartTrainingPayload.ts
export interface TrainingEventOptions {
  progressSampleInterval?: number;
  outputSampleInterval?: number;
  weightsSampleInterval?: number;
  progressMinMillis?: number;
  weightsMinMillis?: number;
}

export interface StartTrainingPayload {
  sessionId: string;
  databaseName: string;
  hiddenLayersNumber: number;
  activationFunctionName: string;
  learningRate: number;
  stopError: number;
  maxEpochs: number;
  eventOptions?: TrainingEventOptions;
}
